app/src/main/java/com/keylesspalace/tusky/MainActivity.java

```diff
@@ -502,24 +499,28 @@ private void updateProfiles() {
 
         List<AccountEntity> allAccounts = accountManager.getAllAccountsOrderedByActive();
 
-        //remove profiles before adding them again to avoid duplicates
-        List<IProfile> profiles = new ArrayList<>(headerResult.getProfiles());
-        for (IProfile profile : profiles) {
-            if (profile.getIdentifier() != DRAWER_ITEM_ADD_ACCOUNT) {
-                headerResult.removeProfile(profile);
+        // reuse the already existing "add account" item
+        List<IProfile> profiles = new ArrayList<>(allAccounts.size()+1);
+        for (IProfile profile: headerResult.getProfiles()) {
+            if (profile.getIdentifier() == DRAWER_ITEM_ADD_ACCOUNT) {
```

Comment:
You could `break` after finding the right profile but shouldn't it be always the last one?